### PR TITLE
Feat/implement hud player zappy gui

### PR DESCRIPTION
**Pull request by mathieurobert1**
I created the Hud for player, Game and Tiles.
When it's free cam :
image
When its pov cam, the stats of the player are displayed.
When its tile cam, the stats of the tile are displayed.

More details : #90

### CODE REVIEW

**Review by Thomaltarix — APPROVED** [review]
Great !

File: gui/include/Render/Render.hpp [hunk 1]
```diff
@@ -94,12 +96,20 @@ class Gui::Render {
         Model                       _thystameModel;     // Model to display thystames.
         Model                       _deraumereModel;    // Model to display deraumeres.
 
+        std::vector<std::shared_ptr<Gui::IHud>>      _hudList;           // List of huds.
```
Comment (Njord201) [inline gui/include/Render/Render.hpp:99]: Put this parameter with the other

**Review by Njord201 — CHANGES_REQUESTED** [review]
One request and try to make tests

File: gui/include/Hud/AHud.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+         * @brief Get the Type object.
+         *
+         * @return TypeScene - Type of the scene.
+         */
+        TypeScene getType();
+
+    protected:
+
+        TypeScene                       _typeScene;     // Type of the scene.
+        std::shared_ptr<GameData>       _gameData;      // GameData class
```
Comment (theophile-jr) [inline gui/include/Hud/AHud.hpp:55]: .

**Review by theophile-jr — CHANGES_REQUESTED** [review]

**Comment by theophile-jr** [comment]
merge w main

**Review by theophile-jr — APPROVED** [review]

File: gui/src/Hud/AHud.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+{
+    _player = player;
+}
+
+void Gui::AHud::setTile(std::shared_ptr<Tile> tile)
+{
+    _tile = tile;
+}
+
+Gui::AHud::TypeScene Gui::AHud::getType()
```
Comment (Njord201) [inline gui/src/Hud/AHud.cpp:20]: Add const label

**Review by Njord201 — CHANGES_REQUESTED** [review]
One request.. good luck :blush:

**Review by Njord201 — COMMENTED** [review]
All's good :+1:

**Review by Njord201 — DISMISSED** [review]

**Review by theophile-jr — APPROVED** [review]

File: gui/src/Hud/HudGame.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+** EPITECH PROJECT, 2024
+** Zappy GUI
+** File description:
+** HudGame
+*/
+
+#include "Assets.hpp"
+#include "Hud/HudGame.hpp"
+
+#include <iostream>
```
Comment (Thomaltarix) [inline gui/src/Hud/HudGame.cpp:11]: It might be debug

**Review by theophile-jr — APPROVED** [review]
LGTM

**Review by Njord201 — DISMISSED** [review]
Great job mister Mathieu!

File: gui/src/Hud/HudTile.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+** EPITECH PROJECT, 2024
+** Zappy GUI
+** File description:
+** HudTile
+*/
+
+#include "Assets.hpp"
+#include "Hud/HudTile.hpp"
+
+#include <iostream>
```
Comment (Thomaltarix) [inline gui/src/Hud/HudTile.cpp:11]: Same i think, it might be debug

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
Just some debug, overall it's a great job

**Review by Njord201 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]
👍

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**